### PR TITLE
Securing Sensu Go

### LINKED ISSUE #849: Guide: Securing Sensu Go
Create a new guide at /guides/securing based on https://docs.sensu.io/sensu-core/1.6/guides/securing-sensu/

Notes:

- etcd peer communication
- Backend API (HTTP)
- Agent API (ws)
- dashboard (HTTP)

Related: #390

### PR DESCRIPTION
**Pull request by asachs01**
Will eventually close #849

### CODE REVIEW

**Comment by asachs01** [comment]
@apaskulin giving a status update here. Still very much a WIP, but on target for pushing out by GA. Given the convo this morning, I need to adjust how the doc has been structured so that it's more in line with how SGO operates.

File: content/sensu-go/5.0/guides/securing-sensu-go.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+---
+title: "How to Secure Sensu Go"
+linkTitle: "Securing Sensu Go"
```
Comment (apaskulin) [inline content/sensu-go/5.0/guides/securing-sensu-go.md]:
```suggestion
```

File: content/sensu-go/5.0/guides/securing-sensu-go.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+---
+title: "How to Secure Sensu Go"
```
Comment (apaskulin) [inline content/sensu-go/5.0/guides/securing-sensu-go.md]:
```suggestion
title: "Securing Sensu"
```

File: content/sensu-go/5.0/guides/securing-sensu-go.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+weight: 12
+version: "5.0"
+product: "Sensu Go"
+platformContent: false
+menu:
+  sensu-go-5.0:
+    parent: guides
+---
+
+As with any piece of software, it is critical to minimize any attack surface exposed by the software. Sensu is no different. If you are familiar with Sensu, you are already aware of the [component pieces that need to be secured][1]. Sensu Go, while different its deployment architecture, still contains component pieces that will need to be secured in order to be considered "production ready":
```
Comment (apaskulin) [inline content/sensu-go/5.0/guides/securing-sensu-go.md]:
```suggestion
As with any piece of software, it is critical to minimize any attack surface exposed by the software. Sensu is no different. The following component pieces need to be secured in order for Sensu to be considered production ready:
```

File: content/sensu-go/5.0/guides/securing-sensu-go.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+* Sensu agent to server communication
+* etcd peer communication
+* Backend API
+* Dashboard
+
+We'll cover securing each one of those pieces, starting with Sensu agent to server communication.
+
+## Securing Sensu agent to server communication
+
+The Sensu agent uses WebSockets for communication between the agent and the server. By default, an agent will use the insecure `ws://` transport:
```
Comment (apaskulin) [inline content/sensu-go/5.0/guides/securing-sensu-go.md]:
```suggestion
The Sensu agent uses WebSockets for communication between the agent and the server. By default, an agent uses the insecure `ws://` transport:
```

File: content/sensu-go/5.0/guides/securing-sensu-go.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+{{< highlight shell >}}
+---
+##
+# agent configuration
+##
+...
+backend-url:
+  - "ws://127.0.0.1:8081"
+{{< /highlight >}}
```
Comment (apaskulin) [inline content/sensu-go/5.0/guides/securing-sensu-go.md]: This sample needs some context: that it's the agent config located at...

File: content/sensu-go/5.0/guides/securing-sensu-go.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+# agent configuration
+##
+...
+backend-url:
+  - "wss://127.0.0.1:8081"
+{{< /highlight >}}
+
+The agent will then connect Sensu servers over wss. Let's move on to securing etcd peer communication.
+
+_NOTE: If creating a Sensu Go cluster, every cluster member will need to be present in the configuration. See the [Sensu Go clustering guide][2] for more information on how to configure agents for a clustered configuration._
```
Comment (apaskulin) [inline content/sensu-go/5.0/guides/securing-sensu-go.md]:
```suggestion
_NOTE: If creating a Sensu cluster, every cluster member needs to be present in the configuration. See the [Sensu Go clustering guide][2] for more information on how to configure agents for a clustered configuration._
```

**Review by apaskulin — COMMENTED** [review]
Super excited to see this coming together! 


Some things to note:
- The filename should be `securing-sensu.md` to match Sensu Core.
- Use "Sensu" instead of "Sensu Go"
- Use "backend" instead of "server"
- Use present tense
- Don't forget to remove the images from this PR

**Review by apaskulin — COMMENTED** [review]
Please update the filename to securing-sensu.md then :shipit:

**Review by apaskulin — APPROVED** [review]

**Pull request merged**